oauthenticator/gitlab.py

```diff
@@ -23,10 +24,26 @@
 
 from .oauth2 import OAuthLoginHandler, OAuthenticator
 
+GITLAB_URL = os.getenv('GITLAB_URL')
+GITLAB_HOST = os.getenv('GITLAB_HOST')
+
+if not GITLAB_URL and GITLAB_HOST:
+    if GITLAB_HOST.startswith('https://') or \
+            GITLAB_HOST.startswith('http://'):
+        warnings.warn('Use GITLAB_URL instead GITLAB_HOST.',
+                      DeprecationWarning)
+        GITLAB_URL = GITLAB_HOST
+    else:
+        GITLAB_URL = 'https://{0}'.format(GITLAB_HOST)
```

Comment:
This is adding a new feature to GITLAB_HOST while deprecating it, which we probably don't need to do. We can just have:

```python
if GITLAB_HOST and not GITLAB_URL:
    warnings.warn(...)
    GITLAB_URL = GITLAB_HOST
```

without looking at the contents of GITLAB_HOST.

Reply:
I will add the warning, but will leave the `https://` prefix. I've burned myself few times by setting `GITLAB_HOST` as host without `https://` prefix, so while the `GITLAB_HOST` is not completely removed I would keep it there for people that don't read docs very carefully ;)

Reply:
In that case, this is adding a *new feature* to fully supported GITLAB_HOST, instead of deprecating it at all. If supporting `GITLAB_HOST='mygitlab.org'` is a desired goal, I would propose this change:

- Don't define `GITLAB_URL`
- just add this `if '://' not in GITLAB_HOST: GITLAB_HOST = "https://GITLAB_HOST"`

I don't think it's useful to have separate environment variables for the same value with/without `https`. If that should be optional, it should be optional, but there is no need for different environment variables where it's optional in one, but not the other.